webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPageActions.js

```diff
@@ -0,0 +1,200 @@
+import history from '../../../history';
+import API from '../../../API';
+import {
+  AUDITS_PAGE_FETCH,
+  AUDITS_PATH,
+  AUDITS_PAGE_SHOW_MESSAGE,
+  AUDITS_PAGE_HIDE_MESSAGE,
+  AUDITS_PAGE_CHANGE_PARAMS,
+} from './AuditsPageConstants';
+import {
+  selectNextPageAudits,
+  selectPrevPageAudits,
+  selectAuditsSelectedPage,
+  selectAuditsShowMessage,
+  selectAuditsPerPage,
+  selectAuditsSearch,
+  selectAudits,
+} from './AuditsPageSelector';
+import { stringifyParams } from '../../../components/Pagination/PaginationHelper';
+import { translate as __ } from '../../../common/I18n';
+
+export const initializeAudits = (params, replace = false) => dispatch => {
+  /**
+   * on didMount, fetch page and page+1
+   */
+  dispatch(fetchAudits(params));
+  dispatch(fetchAudits({ ...params, page: params.page + 1 }, 'nextAudits'));
+  if (replace)
+    history.replace({
+      pathname: AUDITS_PATH,
+      search: stringifyParams(params),
+    });
+  dispatch({
+    type: AUDITS_PAGE_CHANGE_PARAMS,
+    payload: {
+      ...params,
+    },
+  });
+};
+
+const fetchAndPush = params => dispatch => {
+  dispatch(fetchAudits(params));
+  dispatch({
+    type: AUDITS_PAGE_CHANGE_PARAMS,
+    payload: {
+      ...params,
+    },
+  });
+  history.push({
+    pathname: AUDITS_PATH,
+    search: stringifyParams(params),
+  });
+};
+export const auditSearch = searchQuery => (dispatch, getState) => {
+  const params = {
+    page: 1,
+    perPage: selectAuditsPerPage(getState()),
+    searchQuery,
+  };
+  dispatch(fetchAndPush(params));
+};
+
+export const changePage = page => (dispatch, getState) => {
+  const params = {
+    page,
+    perPage: selectAuditsPerPage(getState()),
```

Comment:
You are using so many selectors here. Does it make sense to put all of them in consts before using them for readability?
```js
const state = getState();
const perPage = selectAuditsPerPage(state);
const searchQuery = selectAuditsSearch(state);
const selectedPage = selectAuditsSelectedPage(state);
...
```